integration/integration.go

```diff
@@ -98,115 +96,51 @@ func New(name, version string, opts ...Option) (i *Integration, err error) {
-	i.locker.Lock()
-	defer i.locker.Unlock()
-
-	for _, e := range i.Entities {
-		if e.isLocalEntity() {
-			return e
-		}
-	}
-
-	e := newLocalEntity(i.storer, i.addHostnameToMeta)
-
-	i.Entities = append(i.Entities, e)
-
-	return e
-}
-
-// EntityReportedBy entity being reported from another entity that is not producing the actual entity data.
-func (i *Integration) EntityReportedBy(reportingEntity EntityKey, reportedEntityName, reportedEntityNamespace string, idAttributes ...IDAttribute) (e *Entity, err error) {
-	e, err = i.Entity(reportedEntityName, reportedEntityNamespace, idAttributes...)
-	if err != nil {
-		return
-	}
-
-	e.setCustomAttribute(AttrReportingEntity, reportingEntity.String())
-	return
-}
+	i.HostEntity = newHostEntity()
 
-// EntityReportedVia entity being reported from a known endpoint.
-func (i *Integration) EntityReportedVia(endpoint, reportedEntityName, reportedEntityNamespace string, idAttributes ...IDAttribute) (e *Entity, err error) {
-	e, err = i.Entity(reportedEntityName, reportedEntityNamespace, idAttributes...)
-	if err != nil {
-		return
-	}
-
-	e.setCustomAttribute(AttrReportingEndpoint, endpoint)
 	return
 }
 
-// Entity method creates or retrieves an already created entity.
-func (i *Integration) Entity(name, namespace string, idAttributes ...IDAttribute) (e *Entity, err error) {
+// NewEntity method creates a new (uniquely named) Entity.
+// The `name` of the Entity must be unique for the account otherwise it will cause conflicts
+func (i *Integration) NewEntity(name string, entityType string, displayName string) (e *Entity, err error) {
 	i.locker.Lock()
 	defer i.locker.Unlock()
 
-	e, err = newEntity(name, namespace, i.storer, i.addHostnameToMeta, idAttributes...)
+	e, err = newEntity(name, entityType, displayName)
 	if err != nil {
 		return nil, err
 	}
 
-	for _, eIt := range i.Entities {
-		if e.SameAs(eIt) {
-			return eIt, nil
-		}
-	}
-
-	defaultArgs := args.GetDefaultArgs(i.args)
-
-	if defaultArgs.Metadata {
-		for _, element := range os.Environ() {
-			variable := strings.Split(element, "=")
-			prefix := fmt.Sprintf("%s%s_", CustomAttrPrefix, strings.ToUpper(i.Name))
-			if strings.HasPrefix(variable[0], prefix) {
-				e.setCustomAttribute(strings.TrimPrefix(variable[0], prefix), variable[1])
-			}
-		}
-	}
+	err = i.addDefaultAttributes(e)
 
-	if defaultArgs.NriCluster != "" {
-		e.setCustomAttribute(CustomAttrCluster, defaultArgs.NriCluster)
-	}
-	if defaultArgs.NriService != "" {
-		e.setCustomAttribute(CustomAttrService, defaultArgs.NriService)
-	}
+	return e, err
+}
 
+// AddEntity adds an entity to the list of entities. No check for "duplicates" is performed
+func (i *Integration) AddEntity(e *Entity) {
 	i.Entities = append(i.Entities, e)
+}
 
-	return e, nil
+// NewEvent creates a new event
+func (i *Integration) NewEvent(timestamp time.Time, summary string, category string) (*event.Event, error) {
+	return event.New(timestamp, summary, category)
```

Comment:
what's the goal?
I guess we should add the event to the integration, otherwise this seems to be just a helper function which I don't think does much besides increasing the API surface

Reply:
fixed